### PR TITLE
Refactorings

### PR DESCRIPTION
**Pull request by JanKanis**
This branch started out as an attempt to refactor some rest-api stuff. I didn't get to that much yet.

Currently this contains some improvements in hypercube code and some additions that will help coming rest-api refactorings. There is also an extra class in rest-api that is not yet used, please ignore that. The main reason to merge now is because of fixed hypercube integration tests that turned out to be doing nothing and failing once the no-ops had been fixed.

I changed the fields that get serialized in SerializableProperties, so this will probably break e2e tests. @BarteldKlasens please update tests.

### CODE REVIEW

**Comment by BarteldKlasens** [comment]
@JanKanis please run the constraintSpec in the json and protobuf packages of transmart-e2e. for most json it returns [httpStatus:500, message:null, type:InvocationTargetException]. For protobuf it sends incomplete messages. Often there is only a header.

File: transmart-rest-api/src/main/groovy/org/transmartproject/rest/protobuf/SerializableProperties.groovy [hunk 1]
```diff
@@ -8,11 +8,11 @@ class SerializableProperties {
...
                                                         "religion", "sourcesystemCd", "sexCd"]
     public static final Set<String> CONCEPT = ["conceptPath", "conceptCode"]
-    public static final Set<String> STUDY = ["studyId", "bioExperimentId", "secureObjectToken"]
-    public static final Set<String> TRIAL_VISIT = ["id", "relTimeUnit", "relTime", "relTimeLabel"]
+    public static final Set<String> STUDY = ["studyId"]
+    public static final Set<String> TRIAL_VISIT = ["relTimeUnit", "relTime", "relTimeLabel"]
     public static final Set<String> START_DATE = ["startDate"]
     public static final Set<String> END_DATE = ["endDate"]
-    public static final Set<String> VISIT = ["encounterNum", "patient", "activeStatusCd", "startDate", "endDate", "inoutCd", "locationCd"]
+    public static final Set<String> VISIT = ["patientInTrialId", "activeStatusCd", "startDate", "endDate", "inoutCd", "locationCd"]
```
Comment (gijskant) [inline transmart-rest-api/src/main/groovy/org/transmartproject/rest/protobuf/SerializableProperties.groovy]: Please add `encounterNum` as well. Currently, this reverts [Fix for TMPDEV-159](https://github.com/thehyve/transmart-upgrade/commit/e60568e1c186cccf57cd6e9eb9952fac01588525)

**Comment by forus** [comment]
@JanKanis 
>There is also an extra class in rest-api that is not yet used, please ignore that. The main reason to merge now is because of fixed hypercube integration tests that turned out to be doing nothing and failing once the no-ops had been fixed.

Would it be much work to split it then?

**Comment by JanKanis** [comment]
@forus Required some advanced git-fu, but I removed (most of) the things I did in rest-api.

**Comment by JanKanis** [comment]
@gijskant Done

**Pull request merged**